src/providers/key/key.ts

```diff
@@ -285,4 +285,8 @@ export class KeyProvider {
       return key1.fingerPrint === key2.fingerPrint;
     else return key1.id === key2.id;
   }
+
+  public getKeyByFingerprint(key) {
```

Comment:
Given "isMatch" will check first fingerPrint and then id, 
maybe a better will be "getMatchedKey"

Reply:
getMatchedKey looks better